common/djangoapps/cme_registration_v2/cme_reg_form/forms.py

```diff
@@ -0,0 +1,270 @@
+import datetime
+
+from django import forms
+
+from cme_reg_form.constants import *
+from cme_reg_form.models import ExtraInfo
+
+
+
+class ExtraInfoForm(forms.ModelForm):
+    """
+    The fields on this form are derived from the ExtraInfo model
+    """
+    def __init__(self, *args, **kwargs):
+        super(ExtraInfoForm, self).__init__(*args, **kwargs)
+        self.fields['birth_date'].initial = '01/01'
+
+    required_css_class = 'required'
+    sub_specialty = forms.ChoiceField(
+        choices=(
+            ('', '---------'),
+        ),
+        required=False,
+    )
+    gender = forms.ChoiceField(
+        choices=(
+            ('', '-----'),
+            ('f', 'Female'),
+            ('m', 'Male'),
+        ),
+        required=False,
+        label='Gender',
+    )
+
+    class Meta(object):
+        model = ExtraInfo
+        fields = (
+            'first_name',
+            'middle_initial',
+            'last_name',
+            'birth_date',
+            'gender',
+            'professional_designation',
+            'license_number',
+            'license_country',
+            'license_state',
+            'physician_status',
+            'patient_population',
+            'specialty',
+            'sub_specialty',
+            'affiliation',
+            # 'other_affiliation', not used?
+            'stanford_department',
+            'sunet_id',
+            'address_1',
+            'address_2',
+            'city',
+            'country',
+            'state',
+            'county_province',
+            'postal_code',
+        )
+        labels = {
+            'county_province': 'International Province or Territory',
+            'postal_code': 'Postal/Zip Code',
+            'full_name': 'CANNOT BE CHANGED LATER',
+            'birth_date': 'Month and Day of Birth',
+            'sunet_id': 'SUnet ID',
+        }
+        error_messages = {
+            'state': {
+                'required': u'Enter your state, son!',
+            },
+            'affiliation': {
+                'required': u'Choose your affiliation',
+            },
+            'middle_initial': {
+                'required': u'Middle initial must be at most 1 character',
```

Comment:
The frontend does enforce this, but we need a validation message for the backend.